akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpMessage.scala

```diff
@@ -353,10 +387,11 @@ final class HttpRequest(
   }
 
   override def equals(obj: scala.Any): Boolean = obj match {
-    case HttpRequest(_method, _uri, _headers, _entity, _protocol) =>
+    case request @ HttpRequest(_method, _uri, _headers, _entity, _protocol) =>
```

Comment:
We cannot add 'attributes' to the unapply in a binary-compatible (neither source-compatible) way, so we have no choice but to have it only as a field. Matching on so many attributes becomes unwieldy anyway, so not too bad.

Reply:
I totally agree.